src/Sylius/Component/Core/Resolver/ShippableCountriesResolver.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Resolver;
+
+use Doctrine\Common\Collections\Collection;
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Resolver\ShippableCountriesResolverInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+final class ShippableCountriesResolver implements ShippableCountriesResolverInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    protected $countryRepository;
+
+    /**
+     * @var ChannelContextInterface
+     */
+    protected $channelContext;
+
+    /**
+     * @param RepositoryInterface $countryRepository
+     * @param ChannelContextInterface $channelContext
+     */
+    public function __construct(RepositoryInterface $countryRepository, ChannelContextInterface $channelContext)
+    {
+        $this->countryRepository = $countryRepository;
+        $this->channelContext = $channelContext;
+    }
+
+    /**
+     * @param ChannelInterface|null $channel
+     *
+     * @return array
+     */
+    public function getShippableCountries(ChannelInterface $channel = null): array
+    {
+        if ($channel === null) {
+            $channel = $this->channelContext->getChannel();
+        }
+
+        $countries = $channel->getShippableCountries();
+
+        if ($countries->isEmpty()) {
+            $countries = $this->countryRepository->findAll();
+        }
+
+        $countries = $countries instanceof Collection ? $countries->getValues() : $countries;
```

Comment:
Do we really need to check that? According to [this typehint](https://github.com/Sylius/Sylius/pull/9142/files#diff-9311628d0c2624bfb9a778b4e09a3d53R314) it is always a `Collection`